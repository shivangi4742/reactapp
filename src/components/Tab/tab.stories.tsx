import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { Card, Grid } from '@material-ui/core';
import Tab from './tab';
import ComponentStories from '../component-stories';
import ComponentAPI from '../component-api';
const stories = storiesOf('Components', module);

const codeString = `
const tabs = ['Daily', 'Weekly', 'Monthly'];

<Tab activeTab={0} handler={1} tabs={tabs} />
`;
const tabs = ['Daily', 'Weekly', 'Monthly'];

stories.add(
    'Tabs',
    (() => <div>
        <h2>Tabs</h2>
        <p>Tabs make it easy to explore and switch between different views.</p>
        <h3>Examples <a href="#examples">#</a></h3>

        <Grid container spacing={40}>
            <ComponentStories syntax={codeString}>
                <Tab activeTab={0} handler={1} tabs={tabs} />
            </ComponentStories>
        </Grid>


    </div>
    )
);