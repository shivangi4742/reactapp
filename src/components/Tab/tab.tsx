import React from 'react';
import Paper from '@material-ui/core/Paper';
import { Tabstyle } from './tab.styles';
import Tabs from '@material-ui/core/Tabs';

export const useTab = (activeTab: number) => {
    const [tabIndex, setTab] = React.useState(activeTab);
    const handleChange = React.useCallback(
        (e, value) => {
            setTab(value);
        },
        [tabIndex]
    );
    return [tabIndex, handleChange];
}

const TmTab = ({ activeTab, handler, tabs }) => {

    const [tabIndex, handleTabChange] = useTab(0);
    return (
        <Paper >
            <Tabs
                value={activeTab}
                onChange={handler}
                indicatorColor="primary"
                textColor="primary"
                centered
            >
                {tabs.map((tab, idx) => <Tabstyle key={idx} label={tab} />)}
            </Tabs>
        </Paper>
    );
}

/* TmTab.defaultProps = {
    activeTab: 0,
    handler: () => {},
    tabs: []
} */

export default TmTab;