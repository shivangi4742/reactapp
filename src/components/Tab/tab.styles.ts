import styled, { css } from 'styled-components';
import DS from '../../utils/design-system';
import Tab from '@material-ui/core/Tab';

export const Tabstyle = styled(Tab)`
    min-width: 100px!important;
   color : ${DS.PALETTE.BLACK};
   font-family: ${DS.FONT_FAMILY.REGULAR};
  font-size: 16px;
  font-weight: 500;
  line-height: 19px;
  text-transform: capitalize !important;
  
`;


