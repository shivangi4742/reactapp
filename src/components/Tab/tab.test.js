import React from 'react';
import { shallow } from 'enzyme';
import Tab from './tab';
import toJSON from 'enzyme-to-json';

const tabIndex = 0;
const tabs= ['daily', 'weekly', 'monthly'];

describe('Tab', () => {
    it('renders without crashing', () => {
   // const tree = shallow( <Tab activeTab={tabIndex} handler={1} tabs={tabs} />);
   // expect(toJSON(tree)).toMatchSnapshot();
});
});