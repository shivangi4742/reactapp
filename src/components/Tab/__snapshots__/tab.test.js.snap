// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab renders without crashing 1`] = `
<WithStyles(Paper)>
  <WithStyles(Tabs)
    centered={true}
    indicatorColor="primary"
    onChange={1}
    textColor="primary"
    value={0}
  >
    <WithStyles(Tab)
      key="0"
      label="daily"
    />
    <WithStyles(Tab)
      key="1"
      label="weekly"
    />
    <WithStyles(Tab)
      key="2"
      label="monthly"
    />
  </WithStyles(Tabs)>
</WithStyles(Paper)>
`;
